src/software/world/ball_model/linear_ball_model.cpp

```diff
@@ -0,0 +1,113 @@
+#include "software/world/ball_model/linear_ball_model.h"
+
+#include "software/new_geom/util/physics.h"
+
+LinearBallModel::LinearBallModel(BallState initial_ball_state,
+                                 std::optional<FrictionParameters> friction_parameters)
+    : initial_ball_state_(initial_ball_state),
+      friction_parameters_(initFrictionParameters(friction_parameters))
+{
```

Comment:
Could we check if the friction parameters are valid and throw an exception if so here? I think it's preferable that we explicitly fail with invalid friction parameters, rather then construct (potentially very invalid) friction parameters by taking the absolute value and carrying on.

Reply:
fixed